Fix LinuxPerfScriptEventParser when parsing timespamp event

gleocadie opened the pull request:
When parsing timestamp of an event ("0.0") with the current culture set to "fr-FR", a System.FormatException is raised.

msftclas commented:
This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

sharwell commented:
Thanks for submitting this! After a quick look at codecov, it appears this line was getting covered by tests prior to your change. To add a test validating your change, a new test could be added next to one of the existing tests that covers the line, and apply the [`UseCulture`](https://github.com/Microsoft/perfview/blob/master/src/PerfView.Tests/StackViewer/StackWindowTests.cs#L22) attribute to force the new test to run with `fr-FR`.

gleocadie commented:
Thanks for your quick answer.
Prior to the PR, I tried to set the culture to "en-US" (using the UseCulture attribute) for one test, but it kept failing. When debugging step by step, I could see the Culture still set to "fr-FR". I dig deeper and noticed that this method was run in another thread (using  a Task) and that would explain why the test is failing. (https://msdn.microsoft.com/fr-fr/library/system.globalization.cultureinfo.currentculture(v=vs.110).aspx)

I'm not sure this test will bring any value, since on a machine with the Culture set to "en-US" with or without the fix, the test still will pass (it may confuse people playing with it).

But if you think it's worth it, I will add the test.

sharwell commented:
> this method was run in another thread

Do you know where exactly this transition occurs? If so, we can [explicitly preserve the culture at the boundary](https://github.com/Microsoft/perfview/blob/316ebcea0ac19b70c89e3b88cdc55654dbb98a08/src/PerfView/GuiUtilities/StatusBar/StatusBar.xaml.cs#L214-L215).

gleocadie commented:
> Do you know where exactly this transition occurs? If so, we can explicitly preserve the culture at the boundary.

Sorry, I removed the link when I cleaned up my message
https://github.com/Microsoft/perfview/blob/9dd9b1acee1f5f35ec28bb220777f81e64de22b8/src/PerfView/OtherSources/Linux/LinuxPerfScriptStackSource.cs#L51

Ok so, I will preserve the culture and add a test.

Thanks for your comment

sharwell commented on src/PerfView/OtherSources/Linux/LinuxPerfScriptStackSource.cs:
```diff
@@ -48,29 +47,45 @@ protected override void DoInterning()
                     threadBlockedTimeAnalyzers[i] = new List<BlockedTimeAnalyzer>();
                 }
 
+                var currentCulture = Thread.CurrentThread.CurrentCulture;
```
:memo: (For reviewers) these changes are more easily reviewed with whitespace ignored:
https://github.com/Microsoft/perfview/pull/311/files?w=1

sharwell commented on src/LinuxEvent.Tests/EventParseTests.cs:
```diff
@@ -95,6 +93,16 @@ public void OneStack()
 		}
 
+        [Fact]
+        [UseCulture("fr-FR")]
```
❓ I'm assuming you ran this test on your machine without your code change, and the test failed due to the described localization problem. Can you confirm that? I will try it locally as well. :smile:

gleocadie replied:
Yes, I ran the test without the change on double.Parse but with the preservation of the culture. The test failed. Setting the culture to **CultureInfo.InvariantCulture** in _double.Parse_ make the test green.
